enterprise/coderd/coderd.go

```diff
@@ -1128,3 +1147,18 @@ func (api *API) runEntitlementsLoop(ctx context.Context) {
 func (api *API) Authorize(r *http.Request, action policy.Action, object rbac.Objecter) bool {
 	return api.AGPL.HTTPAuth.Authorize(r, action, object)
 }
+
+// nolint:revive // featureEnabled is a legit control flag.
+func (api *API) setupPrebuilds(featureEnabled bool) (agplprebuilds.ReconciliationOrchestrator, agplprebuilds.Claimer) {
+	experimentEnabled := api.AGPL.Experiments.Enabled(codersdk.ExperimentWorkspacePrebuilds)
+	if !experimentEnabled || !featureEnabled {
+		api.Logger.Debug(context.Background(), "prebuilds not enabled",
+			slog.F("experiment_enabled", experimentEnabled), slog.F("feature_enabled", featureEnabled))
+
+		return agplprebuilds.DefaultReconciler, agplprebuilds.DefaultClaimer
+	}
```

Comment:
It might be worth having an clearly differentiated error log if the entitlement is missing?

Reply:
Good idea 👍